cli/pkg/update/update.go

```diff
@@ -0,0 +1,162 @@
+package update
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/hashicorp/go-version"
+	"github.com/rilldata/rill/cli/pkg/dotrill"
+)
+
+const addr = "https://api.github.com/repos/rilldata/rill-developer/releases/latest"
+
+func CheckVersion(ctx context.Context, currentVersion string) error {
+	// Check if build from source
+	if currentVersion == "" {
+		return nil
+	}
+
+	latestVersion, err := checkVersion(ctx)
+	if err != nil {
+		return err
+	}
+
+	v1, err := version.NewVersion(currentVersion)
+	if err != nil {
+		return err
+	}
+
+	v2, err := version.NewVersion(latestVersion)
+	if err != nil {
+		return err
+	}
+
+	if v1.LessThan(v2) {
+		fmt.Printf("\n%s %s → %s\n\n",
+			color.YellowString("A new version of rill is available:"),
+			color.CyanString(currentVersion),
+			color.CyanString(latestVersion))
+		return nil
+	}
+
+	return nil
+}
+
+func checkVersion(ctx context.Context) (string, error) {
+	latestVersion, err := LatestVersion(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	err = dotrill.SetVersionUpdatedAt(time.Now().Format("2006-01-02 15:04"))
+	if err != nil {
+		return "", err
+	}
+
+	err = dotrill.SetVersion(latestVersion)
+	if err != nil {
+		return "", err
+	}
+
+	return latestVersion, nil
+}
```

Comment:
Move the calls to `SetXXX` into `LatestVersion`, and remove this function (just call `LatestVersion` directly from `CheckVersion`